presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHiveMetastoreAuthentication.java

```diff
@@ -44,6 +45,28 @@
     public KerberosHiveMetastoreAuthentication(HiveClientConfig hiveClientConfig, @ForHiveMetastore HadoopAuthentication authentication)
     {
         this(hiveClientConfig.getHiveMetastoreServicePrincipal(), authentication);
+        verifyKerberosProperties(hiveClientConfig);
+    }
+
+    private static void verifyKerberosProperties(HiveClientConfig hiveClientConfig)
+    {
+        requireNonNull(hiveClientConfig.getHiveMetastoreServicePrincipal(),
+                "Kerberos principal is not set for metastore service");
+
+        requireNonNull(hiveClientConfig.getHiveMetastoreClientPrincipal(),
+                "Kerberos principal is not set for metastore client");
+
+        requireNonNull(hiveClientConfig.getHiveMetastoreClientKeytab(),
+                "Kerberos keytab is not set for metastore client");
+
+        requireNonNull(hiveClientConfig.getHdfsPrestoPrincipal(),
+                "Kerberos principal is not set for HDFS");
+
+        requireNonNull(hiveClientConfig.getHdfsPrestoKeytab(),
+                "Kerberos keytab is not set for HDFS");
+
+        checkState(HdfsAuthenticationType.KERBEROS == hiveClientConfig.getHdfsAuthenticationType(),
```

Comment:
This should be `checkArgument` since it's validating an argument. Also, flip the equality so the constant is on the left (which is the canonical way to write it).